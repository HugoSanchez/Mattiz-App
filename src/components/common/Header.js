import React from 'react';
import { View, Text, Image } from 'react-native';
import Icon from 'react-native-vector-icons/SimpleLineIcons';

import colors from '../../constants/colors'

// Props:

const Header = props => {
    return (
        <View style={[ styles.container, props.enlargedElement ? {height: '15%'} : null ]}>
            <View style={{ flexDirection: 'row'}}>
            <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center'}}>
                {
                    props.leftIcon ?
                    <Icon name={props.leftIcon} size={20} color={colors.primaryBlue} style={{ marginTop: '35%' }}/> 
                    :
                    null
                }
            </View>
            <View style={{ flex: 2, justifyContent: 'center', alignItems: 'center'}}>
                <Image 
                    style={{ height: 35, width: 70, marginTop: '20%'}}
                    source={require('../../assets/doubleLeaf.png')}
                />
            </View>
            <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center'}}>
                <Icon name={'bell'} size={20} color={colors.primaryBlue} style={{ marginTop: '40%' }}/> 
            </View>
            </View>
            {
                props.enlargedElement ? 
                props.enlargedElement
                :
                null
            }
        </View>
    );
}

const styles = {
    container: {
        width: '100%',
        height: '12%',
        backgroundColor: '#fff', 
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 5 },
        shadowOpacity: 0.29,
        shadowRadius: 4.65,
        elevation: 2,
    }
};

export default Header;