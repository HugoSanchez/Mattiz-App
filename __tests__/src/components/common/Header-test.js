import 'react-native'
import React from 'react'
import Header from '../../../../src/components/common/Header'
import {render, fireEvent} from '@testing-library/react-native'

test('Renders with title', () => {
	const {container} = render(<Header />)
	expect(container.children[0]).toMatchSnapshot()
})
