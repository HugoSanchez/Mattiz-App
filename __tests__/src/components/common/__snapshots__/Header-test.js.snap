// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "height": "12%",
        "paddingTop": "3%",
        "width": "100%",
      },
      null,
      null,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": "1%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/menu.png",
          }
        }
        style={
          Object {
            "alignSelf": "flex-start",
            "height": 30,
            "marginLeft": "20%",
            "marginTop": "35%",
            "width": 30,
          }
        }
      />
    </TouchableOpacity>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
          "marginTop": "7%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#154053",
            "fontFamily": "Didot-Bold",
            "fontSize": 18,
          }
        }
      >
        MATTIZ!
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": "1%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#154053",
              "fontSize": 20,
            },
            Object {
              "alignSelf": "flex-end",
              "marginRight": "20%",
              "marginTop": "40%",
            },
            Object {
              "fontFamily": "simple-line-icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î€§
      </Text>
    </View>
  </View>
</View>
`;
